src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,22 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_CANDIDATE_AGE = 35;
+    private static final String REQUIRED_CANDIDATE_NATIONALITY = "Ukrainian";
+    private static final int REQUIRED_CANDIDATE_PERIOD_IN_UKRAINE = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate.getAge() >= MIN_CANDIDATE_AGE
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals(REQUIRED_CANDIDATE_NATIONALITY)
+                && (Integer.parseInt(candidate.getPeriodsInUkr()
+                .split("-")[1])
+                - (Integer.parseInt(candidate.getPeriodsInUkr()
+                .split("-")[0])))
```

Comment:
Create method var that will contain result of split instead of do it every rime
```suggestion
                && (Integer.parseInt(period[1])
                - (Integer.parseInt(period[0])))
```